ta4j-core/src/main/java/org/ta4j/core/BaseBar.java

```diff
@@ -230,23 +272,19 @@ public void addTrade(Num tradeVolume, Num tradePrice) {
     }
 
     @Override
-    public void addPrice(Num price) {
-        if (openPrice == null) {
-            openPrice = price;
-        }
+	public void addPrice(Num price) {
+		if (openPrice == null) {
+			openPrice = price;
+		}
 
-        closePrice = price;
-        if (maxPrice == null) {
-            maxPrice = price;
-        } else if(maxPrice.isLessThan(price)) {
-            maxPrice = price;
-        }
-        if (minPrice == null) {
-            minPrice = price;
-        } else if(minPrice.isGreaterThan(price)){
-            minPrice = price;
-        }
-    }
+		closePrice = price;
+		if (maxPrice == null || maxPrice.isLessThan(price)) {
```

Comment:
Please sync your code with current ta4j develop branch. maxPrice and minPrice do not exists